python/cuml/dask/common/dask_df_utils.py

```diff
@@ -54,6 +57,45 @@ def extract_ddf_partitions(ddf, client=None, agg=True):
     yield wait(worker_to_parts)
     raise gen.Return(worker_to_parts)
 
+@gen.coroutine
+def extract_colocated_ddf_partitions(X_ddf, y_ddf, client=None):
+    """
+    Given Dask cuDF input X and y, return an OrderedDict mapping
+    'worker -> [list of futures]' with the enforced co-locality
+     for each partition in ddf.
+
+    :param X_ddf: Dask.dataframe split dataframe partitions into a list of
+               futures.
+    :param y_ddf: Dask.dataframe split dataframe partitions into a list of
+               futures.
+    :param client: dask.distributed.Client Optional client to use
+    """
+    client = default_client() if client is None else client
+    data_parts = X_ddf.to_delayed()
+    label_parts = y_ddf.to_delayed()
+
+    if isinstance(data_parts, np.ndarray):
+        assert data_parts.shape[1] == 1
+        data_parts = data_parts.flatten().tolist()
+
+    if isinstance(label_parts, np.ndarray):
+        assert label_parts.ndim == 1 or label_parts.shape[1] == 1
+        label_parts = label_parts.flatten().tolist()
+
+    parts = list(map(delayed, zip(data_parts, label_parts)))
+    parts = client.compute(parts) 
+    yield wait(parts)
```

Comment:
This seems fine to me, but it would be good to get input from @cjnolet too as he would be more knowledgeable here than I am.

Reply:
Got this directly from XGBoost. So don't know what's good or bad :).

Reply:
I guess @cjnolet said LGTM on slack for this.